Error while checking build/O2/fullCI for 8a18c306b233605b8dbb925a26652b957b7c7de8 at 2023-11-23 17:15:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/QualityControl-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
command /sw/slc8_x86-64/O2/12306-slc8_x86-64-local1/prodtests/full-system-test/dpl-workflow.sh had nonzero exit code 1
[ERROR] invalid workflow in o2-zdc-tdccalib-workflow: Input <matcher query: (and origin:ZDC (and description:TDC_1DH (just startTime:$0 )))> of zdc-tdccalib has lifetime Timeframe, but output ZDC/TDC_1DH of zdc-tdccalib-epn has lifetime Sporadic
[ERROR] Input ZDC/TDCCALIBDATA/0 of zdc-tdccalib has lifetime Timeframe, but output ZDC/TDCCALIBDATA/0 of zdc-tdccalib-epn has lifetime Sporadic
[ERROR] 


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/fc968abf0faaf14ed4bca0ff8c12c8608677014c/slc8_x86-64/o2checkcode/1.0-local216/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12306/8a18c306b233605b8dbb925a26652b957b7c7de8/build_O2_fullCI/pretty.html).

--- a/Framework/Core/src/WorkflowHelpers.cxx
+++ b/Framework/Core/src/WorkflowHelpers.cxx
@@ -1232,6 +1232,17 @@ std::vector<InputSpec> WorkflowHelpers::computeDanglingOutputs(WorkflowSpec cons
   return results;
 }
 
+bool validateLifetime(std::ostream& errors, DataProcessorSpec const& producer, OutputSpec const& output, DataProcessorSpec const& consumer, InputSpec const& input)
+{
+  if (input.lifetime == Lifetime::Timeframe && output.lifetime == Lifetime::Sporadic) {
+    errors << fmt::format("Input {} of {} has lifetime Timeframe, but output {} of {} has lifetime Sporadic\n",
+                          DataSpecUtils::describe(input).c_str(), consumer.name,
+                          DataSpecUtils::describe(output).c_str(), producer.name);
+    return false;
+  }
+  return true;
+}
+
 bool validateExpendable(std::ostream& errors, DataProcessorSpec const& producer, OutputSpec const& output, DataProcessorSpec const& consumer, InputSpec const& input)
 {
   auto isExpendable = [](DataProcessorLabel const& label) {
@@ -1256,7 +1267,7 @@ void WorkflowHelpers::validateEdges(WorkflowSpec const& workflow,
                                     std::vector<DeviceConnectionEdge> const& edges,
                                     std::vector<OutputSpec> const& outputs)
 {
-  std::vector<Validator> defaultValidators = {validateExpendable};
+  std::vector<Validator> defaultValidators = {validateExpendable, validateLifetime};
   std::stringstream errors;
   // Iterate over all the edges.
   // Get the input lifetime and the output lifetime.

